src/api.ts

```diff
@@ -101,14 +106,15 @@ export class JamsocketApi {
     return `https://app.${hostname}/cli-login`
   }
 
-  private async makeRequest(endpoint: string, method: HttpMethod, body?: any, headers?: Headers): Promise<any> {
+  private async makeRequest<T>(endpoint: string, method: HttpMethod, body?: any, headers?: Headers): Promise<T> {
     const url = `${this.apiBase}${endpoint}`
     const response = await request(url, body || null, { ...this.options, method, headers })
 
     const isJSONContentType = response.headers['content-type'] === 'application/json'
     let responseBody
     try {
       responseBody = JSON.parse(response.body)
+      // TODO: add a runtime check here to make sure the response matches the type T
```

Comment:
Is this even possible in typescript? The typescript compiler will strip out type info so that it's not available, right?

Reply:
Not sure if it's possible with generic types, but it might make sense to write runtime checks for each individual type so we can have type safety _and_ use `JSON.parse`.